src/server/project.ts

```diff
@@ -1901,16 +1901,20 @@ namespace ts.server {
                     compilerOptions,
                     moduleResolutionHost));
 
+                const program = hostProject.getCurrentProgram()!;
+                const symlinkCache = hostProject.getSymlinkCache();
                 for (const resolution of resolutions) {
                     if (!resolution.resolvedTypeReferenceDirective?.resolvedFileName) continue;
-                    const { resolvedFileName } = resolution.resolvedTypeReferenceDirective;
-                    const fileName = moduleResolutionHost.realpath?.(resolvedFileName) || resolvedFileName;
-                    if (!hostProject.getCurrentProgram()!.getSourceFile(fileName) && !hostProject.getCurrentProgram()!.getSourceFile(resolvedFileName)) {
-                        rootNames = append(rootNames, fileName);
+                    const { resolvedFileName, originalFileName } = resolution.resolvedTypeReferenceDirective;
+                    if ((!originalFileName || !program.getSourceFile(originalFileName)) && !program.getSourceFile(resolvedFileName)) {

```

Comment:
Given by default we dont have preserveSymlink set to true, do we want to first check `program.getSourceFile(resolvedFileName)` before doing anything with originalFileName?